From: Jean Sacren <sakiwit@gmail.com>
Date: Tue, 12 Oct 2021 01:46:45 -0600
Subject: net: qed_debug: fix check of false (grc_param < 0) expression
Patch-mainline: v5.16-rc1
Git-commit: 50515cac8d0ea2026db2f0cb456bfa51d5e8e85f
References: jsc#SLE-19001

The type of enum dbg_grc_params has the enumerator list starting from 0.
When grc_param is declared by enum dbg_grc_params, (grc_param < 0) is
always false.  We should remove the check of this expression.

Signed-off-by: Jean Sacren <sakiwit@gmail.com>
Acked-by: Shai Malin <smalin@marvell.com>
Link: https://lore.kernel.org/r/20211012074645.12864-1-sakiwit@gmail.com
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/qlogic/qed/qed_debug.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/qlogic/qed/qed_debug.c
+++ b/drivers/net/ethernet/qlogic/qed/qed_debug.c
@@ -5256,7 +5256,7 @@ enum dbg_status qed_dbg_grc_config(struc
 	 */
 	qed_dbg_grc_init_params(p_hwfn);
 
-	if (grc_param >= MAX_DBG_GRC_PARAMS || grc_param < 0)
+	if (grc_param >= MAX_DBG_GRC_PARAMS)
 		return DBG_STATUS_INVALID_ARGS;
 	if (val < s_grc_param_defs[grc_param].min ||
 	    val > s_grc_param_defs[grc_param].max)
